pandas/tests/indexes/ranges/test_join.py

```diff
@@ -0,0 +1,174 @@
+import numpy as np
+
+from pandas import Index, Int64Index, RangeIndex
+import pandas._testing as tm
+
+
+class TestJoin:
```

Comment:
Is there a reason this is OOP?  We should try to be functional if possible.

Reply:
i still find classes useful for test organization